modules/core/src/main/scala/gem/math/Angle.scala

```diff
@@ -0,0 +1,262 @@
+   */
+  def toHourAngle: HourAngle =
+    HourAngle.fromMilliseconds(toMilliarcseconds.toLong / 15L)
+
+  /**
+   * Convert to the closest hour angle iff its magnitide is an even multiple of 15 milliarcseconds.
+   * Exact and invertible where defined.
+   */
+  @SuppressWarnings(Array("org.wartremover.warts.Equals"))
+  def toHourAngleExact: Option[HourAngle] =
+    if (toMilliarcseconds % 15 == 0) Some(toHourAngle) else None
+
+  /**
+   * Destructure this value into a sum of degrees, arcminutes, arcseconds, and milliarcseconds.
+   * Exact, invertible via `Angle.fromDMS`.
+   */
+  def toDMS: Angle.DMS =
+    Angle.DMS(this)
+
+  /** Sum of this angle and `a`. Exact, commutative, invertible. */
+  def +(a: Angle): Angle =
+    Angle.fromMilliarcseconds(toMilliarcseconds.toLong + a.toMilliarcseconds.toLong)
+
+  /** Difference of this angle and `a`. Exact, invertible. */
+  def -(a: Angle): Angle =
+    Angle.fromMilliarcseconds(toMilliarcseconds.toLong - a.toMilliarcseconds.toLong)
+
+  /** String representation of this Angle, for debugging purposes only. */
+  override def toString =
+    f"Angle($toDMS, $toDoubleDegrees°)"
+
+  /** Angles are equal if their magnitudes are equal. Exact. */
+  @SuppressWarnings(Array("org.wartremover.warts.Equals"))
+  override final def equals(a: Any) =
+    a match {
+      case a: Angle => a.toMilliarcseconds == toMilliarcseconds
+      case _        => false
+    }
+
+  override final def hashCode =
+    toMilliarcseconds
+
+}
+
+object Angle {
+
+  val Angle0:   Angle = fromDegrees(0L)
+  val Angle90:  Angle = fromDegrees(90L)
+  val Angle180: Angle = fromDegrees(180L)
+
+  /** Construct a new Angle of the given magnitide in integral milliarcseconds, modulo 360°. Exact. */
+  def fromMilliarcseconds(mas: Long): Angle = {
+    val masPer360 = 360 * 60 * 60 * 1000
+    val masʹ = (((mas % masPer360) + masPer360) % masPer360)
+    new Angle(masʹ.toInt)
+  }
+
+  /** Construct a new Angle of the given magnitide in integral arcseconds, modulo 360. Exact. */
+  def fromArcseconds(as: Long): Angle = fromMilliarcseconds(as * 1000)
+
+  /** Construct a new Angle of the given magnitide in integral arcminutes, modulo 360. Exact. */
+  def fromArcminutes(ms: Long): Angle = fromArcseconds(ms * 60)
+
+  /** Construct a new Angle of the given magnitide in integral degrees, modulo 360. Exact. */
+  def fromDegrees(ms: Long):    Angle = fromArcminutes(ms * 60)
+
+  /** Construct a new Angle of the given magnitide in integral degrees, modulo 360°. Approximate. */
+  def fromDoubleDegrees(ds: Double): Angle =
+    fromMilliarcseconds((ds * 60 * 60 * 1000).toLong)
+
+  /** Construct a new Angle of the given magnitide in radians, modulo 2π. Approximate. */
+  def fromDoubleRadians(rad: Double): Angle =
+    fromDoubleDegrees(rad.toDegrees)
+
+  /** Angle is an Abelian group, but monoid is the best we can do for now. */
+  implicit val AngleMonoid: Monoid[Angle] =
+    Monoid.instance(_ + _, Angle.Angle0)
```

Comment:
I guess they can't be multiplied but I'm asking because I never worked with angles before and don't know what is usually done with them. Is summing angles the only monoidal operation you would do with 2 angles?